tutorial/markdown/golang/quickstart-golang-gin.md

```diff
@@ -21,252 +20,563 @@ sdk_language:
+
+Let us look at the query used here:
+
+```go
+query := `
+SELECT DISTINCT route.destinationairport
+FROM airport AS airport
+JOIN route AS route ON route.sourceairport = airport.faa
+WHERE airport.faa = $airport AND route.stops = 0
+ORDER BY route.destinationairport
+LIMIT $limit
+OFFSET $offset
+`
+
+params := map[string]interface{}{
+  "airport": airport,
+  "limit":   limit,
+  "offset":  offset,
+}
+
+// Use the method to execute the query and return the results
+queryResult, err := ac.AirportService.QueryDirectConnectionAirport(query, params)
+```
 
-from SearchProfile method of controllers/profile_controller.go
+The `QueryDirectConnectionAirport` method returns a slice of `models.Destination` containing the result documents obtained from the query.
 
-### Running The Tests
+We are fetching the direct connections by joining the airport collection with the route collection and filtering based on the source airport specified by the user and by routes with no stops.
+## Running Tests
 
-To run the standard integration tests, use the following commands from the src directory:
+We have defined integration tests using the standard go [testing package](https://pkg.go.dev/testing) for all the API end points. The integration tests use the same database configuration as the application. For the unit tests, we perform the operation using the API and confirm the results by checking the documents in the database. For example, to check the creation of the document by the API, we would call the API to create the document and then read the same document from the database and compare them. After the tests, the documents are cleaned up by calling the DELETE endpoint.
 
-```bash
+```sh
+# Run standard tests for the project
 cd test
 go test -v
 ```
+## Appendix
+
+### Extending API by Adding New Entity
+
+If you would like to add another entity to the APIs, these are the steps to follow:
+
+- **Create the New Entity in Couchbase Bucket:**
+  - Utilize the [SDK](https://pkg.go.dev/github.com/couchbase/go-couchbase#Bucket.CreateCollection) or the [Couchbase Server interface](https://docs.couchbase.com/cloud/n1ql/n1ql-language-reference/createcollection.html) to establish the new collection within the Couchbase bucket.
+
+- **Define the New Route:**
+  - Navigate to the `route/routes.go` folder and create the new route.
+
+- **Controller Configuration:**
+  - Develop a new file in the `controllers` folder, mirroring the existing structures (e.g., `airport_controller.go`). Craft the corresponding method within this file to manage the new entity.
+
+- **Service Layer Implementation:**
+  - Generate a new file in the `service` folder, modeling it after the existing service files (e.g., `airline.go`). Construct the service logic pertinent to the operations involving the new entity.
```

Comment:
Can you change `airline.go` to `airport.go` so that it is the same entity everywhere?

Reply:
Changed to airport.go